### PR TITLE
[BEAM-6211] Support deflate (zlib) in CompressedFile

### PR DESCRIPTION
**Pull request by brianmartin**
`.deflate` files are quite common in Hadoop and also supported by TensorFlow in TFRecord file format. Moreover, `.deflate` is already supported since 0.6.0 by the Java SDK (see BEAM-1518).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_GradleBuild/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_GradleBuild/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Gradle/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza_Gradle/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark_Gradle/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by brianmartin** [comment]
Hey @chamikaramj, let me know if there's anything I can do to help. Thanks!

**Comment by iemejia** [comment]
It seems this needs a rebase @brianmartin. Pinging @chamikaramj to maybe start the review of this one since it has been opened since almost a month.

**Comment by brianmartin** [comment]
Thanks for the bump @iemejia! Do you think there's any chance of being able to get this into 2.10.x? I see the first RC has already been cut a couple days ago.

**Comment by chamikaramj** [comment]
This fell out of my radar. Will review in a bit. Thanks.

**Comment by brianmartin** [comment]
Thanks @chamikaramj and no worries. I just did the rebase and will be watching to see if the preliminary build is still passing.

File: sdks/python/apache_beam/io/textio_test.py [hunk 1]
```diff
@@ -641,6 +654,82 @@ def test_read_bzip2_concat(self):
...
+        f.write(zlib.compress('\n'.join(lines)))
+
+      pipeline = TestPipeline()
+      pcoll = pipeline | 'Read' >> ReadFromText(
+          file_name,
+          0, CompressionTypes.DEFLATE,
+          True, coders.StrUtf8Coder())
+      assert_that(pcoll, equal_to(lines))
+      pipeline.run()
+
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/textio_test.py]: Please add tests for text sink (textio.WriteToText) as well.

Comment (brianmartin) [reply sdks/python/apache_beam/io/textio_test.py]: +1 Thanks for catching that. Added.

**Comment by chamikaramj** [comment]
Run Python PreCommit

**Comment by brianmartin** [comment]
I wouldn't expect it's related. Is that gcsio test a bit flaky?

**Comment by chamikaramj** [comment]
Thanks. Looks like all tests passed. Merging.

**Pull request merged**